# Czyszczenie danych i przygotowanie zbioru Titanic do analizy

1. Załadowanie zbioru danych Titanic.

In [310]:
import pandas as pd

In [311]:
data = pd.read_csv ("C:/Users/Laptop/Documents/TitanicMess.tsv", sep = '\t')

2. Wykonanie funkcji head(), która wyświetla 5 pierwszych wierszy ze zbioru danych.

In [312]:
data.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38      1   
2                             Heikkinen, Miss. Laina  female  26      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35      1   
4                           Allen, Mr. William Henry    male  35      0   

   Parch            Ticket     Fare Cabin Embarked     ship  
0      0         A/5 21171     7,25   NaN        S  Titanic  
1      0          PC 17599  71,2833   C85        C  Titanic  
2      0  STON/O2. 3101282    7,925   NaN        S  Titanic  
3      0            113803     53,1  C123        S  Titanic  
4      0            373450     8,05   NaN        S  Titanic

Analizując zbiór danych Titanic możemy wyróżnić następujące nazwy kolumn:
PassengerId - unikalny identyfikator,
Survived - pasażer przeżył(1) bądź zmarł(0),
Pclass - klasa, w której podróżował pasażer, możliwe wartości to: 1, 2, 3,
Name - imię pasażera,
Sex - płeć pasażera,
Age - wiek pasażera,
SibSp - liczba rodzeństwa lub obecność małżonka,
Parch - liczba rodziców lub dzieci,
Ticket - numer biletu,
Fare - cena biletu,
Cabin - numer kabiny,
Embarked - port wypłynięcia pasażera, możliwe wartości to: C = Cherbourg; Q = Queenstown; S = Southampton,
ship - nazwa statku.

3. Użycie funkcji info() do wyświetlenia informacji o formacie danych.

In [313]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 892 entries, 0 to 891
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   PassengerId  892 non-null    int64 
 1   Survived     892 non-null    int64 
 2   Pclass       892 non-null    int64 
 3   Name         892 non-null    object
 4   Sex          892 non-null    object
 5   Age          719 non-null    object
 6   SibSp        892 non-null    int64 
 7   Parch        892 non-null    int64 
 8   Ticket       892 non-null    object
 9   Fare         892 non-null    object
 10  Cabin        207 non-null    object
 11  Embarked     890 non-null    object
 12  ship         892 non-null    object
dtypes: int64(5), object(8)
memory usage: 90.7+ KB


Z powyższego zestawienia możemy zobaczyć, że w zbiorze danych znajdują się 892 rekordy. Jako pierwsze rzuca nam się wystąpienie wartości pustych w kolumnie Age, Cabin oraz Embarked. 

4. Dane brakujące.

In [314]:
data['Age'].isnull().sum()

173

In [315]:
data['Cabin'].isnull().sum()

685

In [316]:
data['Embarked'].isnull().sum()

2

Dzięki powyższym funkcją dowiedzieliśmy się o konkretnej liczbie brakujących wartości w poprzednio wskazanych kolumnach. Ponieważ brakujących wartości w kolumnie "Embarked" jest zaledwie 0,2% to zdecydujemy się na usunięcie tych wierszy. Kolumna "Cabin" nie dostarcza nam żadnej pożytecznej wiedzy, posiada aż 76% wartości pustych, dlatego decydujemy się na jej całkowite usunięcie, ponieważ jest bezużyteczna i wpłynie na każdy model uczenia maszynowego. Jeśli chodzi o kolumnę "Age" usunięcie wartości pustych oraz inne rozwiązania problemów zostaną opisane poniżej. Dodatkowo zostanie przez nas usunięta cała kolumna "ship", ponieważ nie spełnia ona żadnej ważnej roli w zbiorze.

In [317]:
data = data.dropna(subset=['Embarked']) #usunięcie pustych wierszy z kolumny Embarked

In [318]:
data['Embarked'].isnull().values.any() #sprawdzenie czy kolumna Embarked posiada wartości puste

False

In [319]:
data = data.drop('Cabin', 1) #usunięcie całej kolumny Cabin

In [320]:
data = data.drop('ship', 1) #usunięcie całej kolumny ship

In [321]:
data.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 890 entries, 0 to 891
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   PassengerId  890 non-null    int64 
 1   Survived     890 non-null    int64 
 2   Pclass       890 non-null    int64 
 3   Name         890 non-null    object
 4   Sex          890 non-null    object
 5   Age          717 non-null    object
 6   SibSp        890 non-null    int64 
 7   Parch        890 non-null    int64 
 8   Ticket       890 non-null    object
 9   Fare         890 non-null    object
 10  Embarked     890 non-null    object
dtypes: int64(5), object(6)
memory usage: 83.4+ KB


5. Przejrzenie wartości w danych kolumnach.

In [322]:
data['Survived'].unique()

array([0, 1], dtype=int64)

In [323]:
data['Pclass'].unique()

array([3, 1, 2], dtype=int64)

In [324]:
data['Name'].unique()

array(['Braund, Mr. Owen Harris',
       'Cumings, Mrs. John Bradley (Florence Briggs Thayer)',
       'Heikkinen, Miss. Laina',
       'Futrelle, Mrs. Jacques Heath (Lily May Peel)',
       'Allen, Mr. William Henry', 'Moran, Mr. James',
       'McCarthy, Mr. Timothy J', 'Palsson, Master. Gosta Leonard',
       'Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)',
       'Nasser, Mrs. Nicholas (Adele Achem)',
       'Sandstrom, Miss. Marguerite Ru&5$$', 'Bonnell, Miss. Elizabeth',
       'Saundercock, Mr. William Henry',
       'Vestrom, Miss. Hulda Amanda Adolfina',
       'Hewlett, Mrs. (Mary D Kingcome)', 'Rice, Master. Eugene',
       'Williams, Mr. Charles Eugene',
       'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)',
       'Masselmani, Mrs. Fatima', 'Fynney, Mr. Joseph J',
       'Beesley, Mr. Lawrence', 'McGowan, Miss. Anna "Annie"',
       'Palsson, Miss. Torborg Danira',
       'Asplund, Mrs. Carl Oscar (Selma Augusta Emilia Johansson)',
       'Emir, Mr. Farred C

In [325]:
data['Sex'].unique()

array(['male', 'female', 'malef', 'mal', 'fem', 'femmale'], dtype=object)

In [326]:
data['Age'].unique()

array(['22', '38', '26', '35', nan, '54', '2', '27', '14', '4', '58',
       '20', '55', '31', '34', '15', '8', '19', '40', '.9', '66', '28',
       '42', '21', '18', '3', '7', '49', '29', '65', '28,5', '5', '11',
       '45', '17', '32', '16', '25', '0,83', '30', '33', '23', '24', '46',
       '59', '71', '37', '47', '14,5', '70,5', '32,5', '12', '9', '36,5',
       '51', '55,5', '40,5', '44', '1', '61', '56', '50', '36', '45,5',
       '4435', '20,5', '62', '41', '52', '63', '23,5', '0,92', '43', '60',
       '39', '10', '64', '13', '48', '0,75', '-3', '-12', '53', '57',
       '80', '250', '70', '24,5', '6', '0,67', '30,5', '0,42', '34,5',
       '74'], dtype=object)

In [327]:
data['SibSp'].unique()

array([1, 0, 3, 4, 2, 5, 8], dtype=int64)

In [328]:
data['Parch'].unique()

array([0, 1, 2, 5, 3, 4], dtype=int64)

In [329]:
data['Ticket'].unique()

array(['A/5 21171', 'PC 17599', 'STON/O2. 3101282', '113803', '373450',
       '330877', '17463', '349909', '347742', '237736', 'PP 9549',
       '113783', 'A/5. 2151', '350406', '248706', '382652', '244373',
       '345763', '2649', '239865', '248698', '330923', '347077', '2631',
       '19950', '330959', '349216', 'PC 17601', 'PC 17569', '335677',
       'C.A. 24579', 'PC 17604', '113789', '2677', 'A./5. 2152', '345764',
       '2651', '7546', '11668', '349253', 'SC/Paris 2123', '330958',
       'S.C./A.4. 23567', '370371', '14311', '2662', '349237', '3101295',
       'A/4. 39886', 'PC 17572', '2926', '113509', '19947', 'C.A. 31026',
       '2697', 'C.A. 34651', 'CA 2144', '2669', '36973', '347088',
       'PC 17605', '2661', 'C.A. 29395', 'S.P. 3464', '3101281', '315151',
       'C.A. 33111', 'S.O.C. 14879', '2680', '1601', '348123', '349208',
       '374746', '248738', '364516', '345767', '345779', '330932',
       '113059', 'SO/C 14885', '3101278', 'W./C. 6608', 'SOTON/OQ 392086',

In [330]:
data['Fare'].unique()

array(['7,25', '71,2833', '7,925', '53,1', '8,05', '8,4583', '51,8625',
       '21,075', '11,1333', '30,0708', '16,7', '26,55', '7,8542', '16',
       '29,125', '13', '18', '7,225', '26', '8,0292', '31,3875', '263',
       '7,8792', '7,8958', '27,7208', '146,5208', '7,75', '10,5',
       '82,1708', '52', '7,2292', '11,2417', '9,475', '21', '41,5792',
       '15,5', '21,6792', '17,8', '39,6875', '7,8', '76,7292', '61,9792',
       '35,5', '27,75', '46,9', '83,475', '27,9', '15,2458', '8,1583',
       '8,6625', '73,5', '14,4542', '56,4958', '7,65', '29', '12,475',
       '9', '9,5', '7,7875', '47,1', '15,85', '34,375', '61,175',
       '20,575', '34,6542', '63,3583', '23', '77,2875', '8,6542', '7,775',
       '24,15', '9,825', '14,4583', '247,5208', '31,275', '7,1417',
       '22,3583', '6,975', '7,05', '14,5', '15,0458', '26,2833', '9,2167',
       '79,2', '6,75', '11,5', '36,75', '7,7958', '12,525', '66,6',
       '7,3125', '61,3792', '7,7333', '69,55', '16,1', '15,75', '20,525',
     

In [331]:
data['Embarked'].unique()

array(['S', 'C', 'Q', 'So', 'Co', 'Qe'], dtype=object)

Podsumowanie analizy:  

Kolumna "Sex" posiada 4 wartości błędne, które powstały w wyniku pomyłki przy uzupełnianiu danych, mamy do czynienia z tzw. literówkami. Naszym rozwiązaniem będzie zamiana błędnych wartości na prawidłowe. 

Przyglądając się wartościom kolumny "Age" możemy zauważyć problem z separatorem, który chcielibyśmy zamienić na separartor ".". Drugą rzeczą jest wystąpienie wartości pustych, rozwiązaniem może okazać się wykorzystanie metody na uzupełnienie pustych danych medianą. Ostatnią rzeczą jaką możemy dostrzec jest wystąpienie wartości ujemnych oraz bardzo dużych liczb.

W kolumnie "Fare" wystąpują przecinki jako separatory, zostaną one poprawione na kropki. Kolumna ta posiada jedną wartość ujemną co jest niedopuszczalne, ponieważ mówimy tutaj o cenie biletu. Rozwiązaniem tego problemu będzie usunięcie całego wiersza z wartością ujemną. Dodatkowo analizując zbiór możemy zauważyć jedną wartość nieliczbową, która zostanie zamieniona na liczbową.

W kolumnie "Embarked" możemy zauważyć zduplikowane wartości, rozwiązaniem będzie sprowadzenie ich tylko do 3 wartości według założeń dotyczących zmiennych w zbiorze danych.

W kolumnie "Ticket" znaleziono wartość "LINE", która nie oznacza numeru biletu zdecydowano na usunięcie wiersza. 

W kolumnie "Name" zauważono niedopuczalne wartości takie jak :"&", "$" oraz cyfry, rozwiązaniem będzie usunięcie tylko tych znaków w wartościach bez konieczności usunięcia całych wierszy.

6. Rozwiązanie dostrzeżonych problemów w zbiorze

Kolumna "Sex"

In [332]:
data.replace({'malef': 'male', 'mal': 'male', 'fem': 'female', 'femmale': 'female'}, inplace=True) #zamiana błędnych wartości na odpowiednie

In [333]:
data['Sex'].unique() #wyświetlenie ponownie wartości w kolumnie Sex

array(['male', 'female'], dtype=object)

Kolumna "Age"

In [334]:
mediana = data['Age'].mode() #za pomocą funkcji mode() odnaleziona zostaje mediana
print('Mediana: ' + str(mediana[0]))
data['Age'] = data['Age'].fillna(str(mediana[0]))
print('Liczba pustych wartości dla Age: ')
print(data['Age'].isnull().sum())

Mediana: 24
Liczba pustych wartości dla Age: 
0


In [335]:
data['Age'] = data['Age'].str.replace(",",".") #zamiana separatora "," na "."

In [336]:
data['Age'] = data['Age'].astype(float) #zmiana typu danych na float

In [337]:
print("Wartość max: " + str(data['Age'].max())) #wyliczenie wartości maksymalnej
print("Wartość min: " + str(data['Age'].min())) #wyliczenie wartości minimalnej

Wartość max: 4435.0
Wartość min: -12.0


Wyświetlony wiek maksymalny i minimalny posiada błędne wartości.

In [338]:
print(data[data['Age']<1][['Age', 'SibSp', 'Parch']]) #wyświetlenie wartości ujemnych
print(data[data['Age']>120][['Age', 'SibSp', 'Parch']]) #wyświetlenie wartości większych niż 120

       Age  SibSp  Parch
31    0.90      1      0
78    0.83      0      2
305   0.92      1      2
469   0.75      2      1
507  -3.00      0      0
522 -12.00      0      0
644   0.75      2      1
755   0.67      1      1
803   0.42      0      1
831   0.83      1      1
        Age  SibSp  Parch
223  4435.0      0      0
666   250.0      0      0


W zbiorze danych możemy zauważyć dwie wartości większe niż 100, dwie wartości ujemne oraz 8 dodatnich, ale mniejszych niż 1. Wartości ujemne i większe niż 120 zostaną zamienione na miedianę. Wartości z przedziału (0, 1) zostaną zamienione na 1.

In [339]:
data.loc[data['Age'] > 120, 'Age'] = float(mediana) # zamiana wieku powyżej 120 na medianę
data.loc[data['Age'] < 0, 'Age'] = float(mediana) # zamiana wieku poniżej 0 na medianę
data.loc[data['Age'] < 1, 'Age'] = float(1) # Zamiana wieku poniżej 1 na 1

print('Wiek mniejszy od 1 lub większy od 120:')
print(data[data['Age']<1][['Age', 'SibSp', 'Parch']])
print(data[data['Age']>120][['Age', 'SibSp', 'Parch']])

Wiek mniejszy od 1 lub większy od 120:
Empty DataFrame
Columns: [Age, SibSp, Parch]
Index: []
Empty DataFrame
Columns: [Age, SibSp, Parch]
Index: []


Kolumna "Fare"

In [340]:
data = data.copy();
data['Fare'] = data['Fare'].str.replace(",",".") # zamiana separatora "," na "."

In [341]:
data['Fare'].unique()

array(['7.25', '71.2833', '7.925', '53.1', '8.05', '8.4583', '51.8625',
       '21.075', '11.1333', '30.0708', '16.7', '26.55', '7.8542', '16',
       '29.125', '13', '18', '7.225', '26', '8.0292', '31.3875', '263',
       '7.8792', '7.8958', '27.7208', '146.5208', '7.75', '10.5',
       '82.1708', '52', '7.2292', '11.2417', '9.475', '21', '41.5792',
       '15.5', '21.6792', '17.8', '39.6875', '7.8', '76.7292', '61.9792',
       '35.5', '27.75', '46.9', '83.475', '27.9', '15.2458', '8.1583',
       '8.6625', '73.5', '14.4542', '56.4958', '7.65', '29', '12.475',
       '9', '9.5', '7.7875', '47.1', '15.85', '34.375', '61.175',
       '20.575', '34.6542', '63.3583', '23', '77.2875', '8.6542', '7.775',
       '24.15', '9.825', '14.4583', '247.5208', '31.275', '7.1417',
       '22.3583', '6.975', '7.05', '14.5', '15.0458', '26.2833', '9.2167',
       '79.2', '6.75', '11.5', '36.75', '7.7958', '12.525', '66.6',
       '7.3125', '61.3792', '7.7333', '69.55', '16.1', '15.75', '20.525',
     

In [342]:
data = data[data.Fare != '-90'] #usunięcie całego wiersza gdzie wartość w kolumnie Fare=-90

In [343]:
data['Fare'].unique() #ponowne wyświetlenie wartości kolumny Fare, możemy zauważyć brak wartości "-90"

array(['7.25', '71.2833', '7.925', '53.1', '8.05', '8.4583', '51.8625',
       '21.075', '11.1333', '30.0708', '16.7', '26.55', '7.8542', '16',
       '29.125', '13', '18', '7.225', '26', '8.0292', '31.3875', '263',
       '7.8792', '7.8958', '27.7208', '146.5208', '7.75', '10.5',
       '82.1708', '52', '7.2292', '11.2417', '9.475', '21', '41.5792',
       '15.5', '21.6792', '17.8', '39.6875', '7.8', '76.7292', '61.9792',
       '35.5', '27.75', '46.9', '83.475', '27.9', '15.2458', '8.1583',
       '8.6625', '73.5', '14.4542', '56.4958', '7.65', '29', '12.475',
       '9', '9.5', '7.7875', '47.1', '15.85', '34.375', '61.175',
       '20.575', '34.6542', '63.3583', '23', '77.2875', '8.6542', '7.775',
       '24.15', '9.825', '14.4583', '247.5208', '31.275', '7.1417',
       '22.3583', '6.975', '7.05', '14.5', '15.0458', '26.2833', '9.2167',
       '79.2', '6.75', '11.5', '36.75', '7.7958', '12.525', '66.6',
       '7.3125', '61.3792', '7.7333', '69.55', '16.1', '15.75', '20.525',
     

In [344]:
data.info() #dla sprawdzenia wyświetlamy ponownie podsumowanie naszego zbioru widzimy, że został "uszczuplony" o 1 wiersz

<class 'pandas.core.frame.DataFrame'>
Int64Index: 889 entries, 0 to 891
Data columns (total 11 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  889 non-null    int64  
 1   Survived     889 non-null    int64  
 2   Pclass       889 non-null    int64  
 3   Name         889 non-null    object 
 4   Sex          889 non-null    object 
 5   Age          889 non-null    float64
 6   SibSp        889 non-null    int64  
 7   Parch        889 non-null    int64  
 8   Ticket       889 non-null    object 
 9   Fare         889 non-null    object 
 10  Embarked     889 non-null    object 
dtypes: float64(1), int64(5), object(5)
memory usage: 83.3+ KB


In [345]:
titanicbf = data['Fare'][data['Fare'].str.contains('[a-zA-Z@_!#$%^&*()<>?/|}{~:]')] #wyszukanie wartości nieliczbowej

print(titanicbf) #wyświetlenie znalezionej wartości nieliczbowej

489    15.9a
Name: Fare, dtype: object


In [346]:
data.loc[data.index == 489, 'Fare'] = '15.9' #zamiana wartości nieliczbowej na wartość liczbową "15,9"

print(data['Fare'][data.index == 489]) #wyświetlenie wartości o wskazanym indeksie

489    15.9
Name: Fare, dtype: object


Kolumna "Embarked"

In [347]:
data.replace({'So': 'S', 'Co': 'C', 'Qe': 'Q'}, inplace=True) #zamiana zduplikowanych wartości

In [348]:
data['Embarked'].unique() #ponowne wyświetlenie wartości w kolumnie Embarked

array(['S', 'C', 'Q'], dtype=object)

Kolumna "Ticket"

In [349]:
data = data[data.Ticket != 'LINE'] #usunięcie wierszy gdzie wartość w kolumnie Ticket=LINE

In [350]:
data['Ticket'].unique() #ponowne wyświetlenie wartości kolumny Ticket, możemy zauważyć brak wartości "LINE"

array(['A/5 21171', 'PC 17599', 'STON/O2. 3101282', '113803', '373450',
       '330877', '17463', '349909', '347742', '237736', 'PP 9549',
       '113783', 'A/5. 2151', '350406', '248706', '382652', '244373',
       '345763', '2649', '239865', '248698', '330923', '347077', '2631',
       '19950', '330959', '349216', 'PC 17601', 'PC 17569', '335677',
       'C.A. 24579', 'PC 17604', '113789', '2677', 'A./5. 2152', '345764',
       '2651', '7546', '11668', '349253', 'SC/Paris 2123', '330958',
       'S.C./A.4. 23567', '370371', '14311', '2662', '349237', '3101295',
       'A/4. 39886', 'PC 17572', '2926', '113509', '19947', 'C.A. 31026',
       '2697', 'C.A. 34651', 'CA 2144', '2669', '36973', '347088',
       'PC 17605', '2661', 'C.A. 29395', 'S.P. 3464', '3101281', '315151',
       'C.A. 33111', 'S.O.C. 14879', '2680', '1601', '348123', '349208',
       '374746', '248738', '364516', '345767', '345779', '330932',
       '113059', 'SO/C 14885', '3101278', 'W./C. 6608', 'SOTON/OQ 392086',

Kolumna "Name"

In [351]:
data = data.copy()
titanicbn = data[data['Name'].str.contains('[@_!#$%^&*<>?/\|}{~:]')] #użycie metody contains() do wyszukania znaków specjalnych

print(titanicbn)

     PassengerId  Survived  Pclass                                Name  \
10            11         1       3  Sandstrom, Miss. Marguerite Ru&5$$   
13            11         1       3  Sandstrom, Miss. Marguerite Ru&5$$   
23            11         1       3  Sandstrom, Miss. Marguerite Ru&5$$   
406          407         0       3    Widegren, Mr. Carl/Charles Peter   

        Sex   Age  SibSp  Parch   Ticket  Fare Embarked  
10   female   4.0      1      1  PP 9549  16.7        S  
13   female   4.0      1      1  PP 9549  16.7        S  
23   female   4.0      1      1  PP 9549  16.7        S  
406    male  51.0      0      0   347064  7.75        S  


Przy wyświetleniu wartości ze znakami specjalnymi z kolumny "Name" zobaczyliśmy, że jeden rekord posiada zdublowane wartości.

In [352]:
data = data.drop_duplicates(inplace=False) #usunięcie duplikatów

print(data['Name'].describe(include=['object']))

count                         882
unique                        882
top       Lefebre, Miss. Mathilde
freq                            1
Name: Name, dtype: object


Ponownie wyświetlamy wartości ze znakami specjalnymi możemy zobaczyć, że problem został usunięty.

In [353]:
data = data.copy()
titanicbn = data[data['Name'].str.contains('[@_!#$%^&*<>?/\|}{~:]')] #użycie metody contains() do wyszukania znaków specjalnych

print(titanicbn)

     PassengerId  Survived  Pclass                                Name  \
10            11         1       3  Sandstrom, Miss. Marguerite Ru&5$$   
406          407         0       3    Widegren, Mr. Carl/Charles Peter   

        Sex   Age  SibSp  Parch   Ticket  Fare Embarked  
10   female   4.0      1      1  PP 9549  16.7        S  
406    male  51.0      0      0   347064  7.75        S  


In [354]:
data['Name'] = data['Name'].str.replace('&', '') #usunięcie "&" z wartości w kolumnie Name

In [355]:
data['Name'] = data['Name'].str.replace('$', '') #usunięcie "$" z wartości w kolumnie Name

In [356]:
data['Name'] = data['Name'].str.replace('5', '') #usunięcie "5" z wartości w kolumnie Name

In [357]:
print(data['Name'][data.index == 10]) #wyświetlenie wiersza o podanym indeksie w celu zweryfikowania zmian

10    Sandstrom, Miss. Marguerite Ru
Name: Name, dtype: object


7. Zapis poprawionych danych do pliku.

In [358]:
titanicc = data.copy();

In [359]:
titanicc.to_csv(r'TitanicCleaned.tsv', sep='\t')

Wykonała: Magda Dytkowicz.